### PR TITLE
new radio fsm

### PR DESCRIPTION
**Pull request by mszczodrak**

### CODE REVIEW

**Comment by mszczodrak** [comment]
This addresses #147

**Comment by mszczodrak** [comment]
@pvanhorn , FYI, here is the OT's new radio FSM

File: examples/platforms/cc2538/radio.c [hunk 1]
```diff
@@ -128,28 +118,49 @@ void setChannel(uint8_t channel)
...
+    return error;
 }
 
 ThreadError otPlatRadioSetExtendedAddress(uint8_t *address)
 {
-    int i;
+    ThreadError error = kThreadError_Busy;
 
-    for (i = 0; i < 8; i++)
+    if (sState != kStateTransmit)
```
Comment (turon) [inline examples/platforms/cc2538/radio.c]: Reversing the logic here simplifies readability:

```
if (sState == kStateTransmit) return kThreadError_Busy;
```

File: include/platform/radio.h [hunk 2]
```diff
@@ -164,7 +189,7 @@ ThreadError otPlatRadioSetShortAddress(uint16_t aShortAddress);
 /**
  * Enable the radio.
  *
- * @retval ::kThreadError_None  Successfully transitioned to Idle.
+ * @retval ::kThreadError_None  Successfully transitioned to Sleep.
```
Comment (turon) [inline include/platform/radio.h]: Doesn't "None" returned by otPlatRadioEnable mean successful transition to "Receive"?

File: examples/platforms/cc2538/radio.c (hunk 1 above)
Comment (mszczodrak) [inline examples/platforms/cc2538/radio.c]: Nope, we want a single return from a function.

File: include/platform/radio.h (hunk 2 above)
Comment (mszczodrak) [inline include/platform/radio.h]: Nope, we transition to Sleep. Please take a look at the FSM description in radio.h

**Comment by turon** [comment]
LGTM

**Comment by jwhui** [comment]
LGTM 👍

**Pull request merged**